domain/ide/workspace/folder_test.go

```diff
@@ -46,21 +47,6 @@ import (
 	"github.com/snyk/snyk-ls/internal/util"
 )
 
-func Test_Scan_WhenCachedResults_shouldNotReScan(t *testing.T) {
-	testutil.UnitTest(t)
-	folderPath, filePath := "testFolderDir", "testPath"
-	scanner := snyk.NewTestScanner()
-
-	scanner.AddTestIssue(NewMockIssue("1", filePath))
-	f := NewFolder(folderPath, "Test", scanner, hover.NewFakeHoverService(), snyk.NewMockScanNotifier(), notification.NewNotifier())
-	ctx := context.Background()
-
-	f.ScanFile(ctx, filePath)
-	f.ScanFile(ctx, filePath)
-
-	assert.Equal(t, 1, scanner.Calls())
-}
-
```

Comment:
I don't understand why and do not agree. WDYT?

Reply:
You do not agree that it should **not** rescan if the results are cached? To me the test makes sense, isn't that the whole point of the caching?

Reply:
Caching makes sense, when we  _access_ information, but not when we try to create. `ScanFile` is supposed to trigger the analysis and the creation of diagnostics. It is e.g. called by `DidSaveTextDocument`. `DidOpenTextDoument` instead only reads the cache, and does not trigger a scan. Stopping a scan because we have cached data is counter-intuitive for me. I'd expect a scan to update the cache, not use it.